src/chainparamsbase.cpp

```diff
@@ -23,6 +24,10 @@ void SetupChainParamsBaseOptions()
     gArgs.AddArg("-segwitheight=<n>", "Set the activation height of segwit. -1 to disable. (regtest-only)", ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::DEBUG_TEST);
     gArgs.AddArg("-testnet", "Use the test chain. Equivalent to -chain=test.", ArgsManager::ALLOW_ANY, OptionsCategory::CHAINPARAMS);
     gArgs.AddArg("-vbparams=deployment:start:end", "Use given start/end times for specified version bits deployment (regtest-only)", ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::CHAINPARAMS);
+    gArgs.AddArg("-signet", "Use the signet chain. Note that the network is defined by the signet_blockscript parameter", ArgsManager::ALLOW_ANY, OptionsCategory::CHAINPARAMS);
+    gArgs.AddArg("-signet_blockscript", "Blocks must satisfy the given script to be considered valid (only for signet networks)", ArgsManager::ALLOW_STRING, OptionsCategory::CHAINPARAMS);
```

Comment:
It seems that this also changes the network magic, so this is a switch to run a different signet?

Reply:
Yep, the blockscript defines the network magic.

https://github.com/bitcoin/bips/blob/master/bip-0325.mediawiki#message-start

Reply:
Wasn't the point of signet to be a single network and everyone could choose their keys to decide which reorg to "follow"?

E.g. there are different keys like:

* no_regorgs_key: signs every block, advancing height by 1 every 10 min, never decreasing height. This is the "true" signet chain
* 10_reorg_key: reorgs 10 blocks every 20 blocks, then reorgs back to to the "true" chain
* 101_reorg_key: reorgs 100 blocks every 200 blocks, then back to the "true" chain

Reply:
That's on the map, for sure, but it adds complexity I'm trying to avoid in the initial release.

As for the single network, yes! The thing is, people are unpredictable, and I can imagine we will end up switching networks on rare occasions; in the end, we may have some form of org handling the privkeys to sign blocks, but initially at least, it will probably be me volunteering to do it, and if I go missing or evil or something, it should be trivial to switch to a new setup (later on, I'd like to make this also possible to switch without having to reset the chain, but that's not MVP stuff). There may also be Bitcoin related companies (e.g. exchanges) who want to have their own signets in addition to the existing one.

The approach here is discussed in https://github.com/bitcoin/bitcoin/pull/16411#issuecomment-577999888 but essentially, we are using the same genesis block for all signets, and instead switching the network magic to avoid mixing signets up.